chrisj117/CS-472-GROUP-2-PROJECT
# Endpoints for Reviews

**_WHAT DOES THIS PR DO?_**
- Consists of endpoints related to `Review`
- [X] Get all reviews
- [X] Get a specific review using the `review_id`
- [X] Get all reviews for a specific school using the `short_name` of the school.
- [X] Create review
- [X] Update review using the `review_id`.
- [X] Delete review using its `review_id`.

**_WHAT ISSUES ARE RELATED TO THIS PR?_**
- [X] #66 
- [X] #67 

**_HOW DO I TEST OUT THIS PR?_**
Run this project locally and test the following endpoints:
Base URL: `http://localhost:8000/api/v1`.
To test the endpoints, go to `http://localhost:8000/api/v1/reviews/` - Use django restframework UI to try out the endpoints
### Testing Endpoints:

- **List all reviews:**  
  `GET /reviews/`  
  Retrieves a list of all reviews.

- **Get a single review by UUID:**  
  `GET /reviews/{review_id}/`  
  Replace `{review_id}` with the actual UUID of the review you want to retrieve.

- **Get reviews for a specific school by its short name:**  
  `GET /school/{school_short_name}/reviews/`  
  Replace `{school_short_name}` with the short name identifier of the school.

- **Create a new review:**  
`POST /reviews/`  
Send a JSON payload with the details of the new review. Ensure the `school` value uses the UUID of an existing school in the database. Example request body:

```json
{
  "school": "27687e0b-98e3-4ea8-98cf-286430552022",
  "review_text": "This is a review.",
  "term": "Fall",
  "grade_received": "A",
  "delivery_method": "Online",
  "helpful_count": 0
}
```
- **Update a review:** 
`PUT /reviews/{review_id}/`
Replace `{review_id}` with the UUID of the review you wish to update. Send a JSON payload with the updated details of the review. Example request body:
```json
{
  "school": "27687e0b-98e3-4ea8-98cf-286430552022",
  "review_text": "Updated review text.",
  "term": "Spring",
  "grade_received": "A-",
  "delivery_method": "In Person",
  "helpful_count": 5
}
```
- **Delete a review:** 
`DELETE /reviews/{review_id}/`
Replace `{review_id}` with the UUID of the review you wish to delete.

NB: Create, Update and Delete  endpoints will later be protected by `JWT` authentication
NB: The `Create a new review` & `Update a review` endpoint currently requires the school's `UUID`. A future update will implement the ability to use the school's short name instead. 


**_ADDITIONAL INFORMATION_**
- [x] Pipeline passing
- [X] Code coverage 100%
- [X] Running locally
